src/expressions/dataTypes/casting/castToStringLikeType.ts

```diff
@@ -14,10 +17,15 @@ export default function castToStringLikeType(
 		});
 	}
 	if (instanceOf('xs:QName')) {
-		return value => ({
-			successful: true,
-			value: value.prefix ? `${value.prefix}:${value.localName}` : value.localName
-		});
+		return value => {
+			const qnameValue = value as QName;
```

Comment:
I think it's better to declare the type of value instead of casting here, i.e. `return (value: QName) => {` and remove `qnameValue` as `value` will now be recognised as a `QName`